pinot-common/src/main/java/org/apache/pinot/pql/parsers/Pql2Compiler.java

```diff
@@ -111,6 +137,64 @@ public BrokerRequest compileToBrokerRequest(String expression)
     }
   }
 
+  private boolean validate(BrokerRequest br1, BrokerRequest br2) throws Exception {
```

Comment:
This could have been a good equals() method on BrokerRequest if it was not auto-generated by thrift. Also, may be compare() is a better name (I understand compare usually returns int, not boolean, but might still be better than validate)?

Reply:
I am using thrift equals methods internally. I just broke it into 3 parts so that we know which part failed. It helped me in debugging. Changed the name to compare